__tests__/app.test.js

```diff
@@ -214,6 +214,43 @@ describe("/api/articles", () => {
         .then(({ body: { msg } }) => expect(msg).toBe("Bad request"));
     });
   });
+
+  describe("Topic query", () => {
+    test("GET: 200 - respond with an array of all article objects when filtered by a existing topic", () => {
+      const topic = "mitch";
+
+      return request(app)
+        .get(`/api/articles?topic=${topic}`)
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toHaveLength(12);
+          articles.forEach((article) =>
+            expect(article).toMatchObject({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: topic,
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+              comment_count: expect.any(Number),
+            })
+          );
+        });
+    });
+
+    test("GET: 200 - respond with an empty array when filtered by a non-existent topic", () => {
+      const topic = "non-existent-topic";
+
+      return request(app)
+        .get(`/api/articles?topic=${topic}`)
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(Array.isArray(articles)).toBe(true);
+          expect(articles).toHaveLength(0);
+        });
+    });
```

Comment:
Don't forget to add in a test for:

- Status 404, non-existent topic query, e.g.` ?topic=bananas`